src/main/scala/uk/gov/nationalarchives/rotate/RotateRealmKeys.scala

```diff
@@ -0,0 +1,70 @@
+package uk.gov.nationalarchives.rotate
+
+import org.keycloak.admin.client.Keycloak
+import org.keycloak.admin.client.resource.ComponentsResource
+import org.keycloak.common.util.MultivaluedHashMap
+import org.keycloak.representations.idm.ComponentRepresentation
+import org.slf4j.Logger
+import org.slf4j.impl.SimpleLoggerFactory
+import uk.gov.nationalarchives.rotate.MessageSender.Message
+
+import scala.jdk.CollectionConverters._
+import scala.util.{Failure, Success, Try}
+
+class RotateRealmKeys(client: Keycloak) {
+  val logger: Logger = new SimpleLoggerFactory().getLogger(this.getClass.getName)
+
+  def rotate(): Message = Try {
+    val realm = client.realm("tdr")
+    val components: ComponentsResource = realm.components()
+    val keyProvider = "org.keycloak.keys.KeyProvider"
+    val keyName = "rsa-generated"
+
+
+    val rsaKeys = components.query("tdr", keyProvider, keyName)
+
+    def filterKeys(filterValue: String): List[ComponentRepresentation] = rsaKeys.asScala.filter(key => key.getConfig.get("active").asScala.toList.head == filterValue).toList
+
+    val inactiveKeys: List[ComponentRepresentation] = filterKeys("false")
+    val activeKeys: List[ComponentRepresentation] = filterKeys("true")
+    //Delete the passive keys from last week, no-one should be using them.
```

Comment:
Maybe reword this comment, in case the schedule for rotating changes: `Delete the previous passive keys, no-one should be using them`